internal/server/grants_test.go

```diff
@@ -565,85 +564,161 @@ func TestAPI_CreateGrant(t *testing.T) {
 		req.Header.Add("Infra-Version", "0.12.3")
 
+		if tc.setup != nil {
+			tc.setup(t, req)
+		}
+
+		resp := httptest.NewRecorder()
 		routes.ServeHTTP(resp, req)
-		assert.Equal(t, resp.Code, http.StatusOK)
 
-		var getGrant api.Grant
-		err = json.NewDecoder(resp.Body).Decode(&getGrant)
-		assert.NilError(t, err)
-		assert.DeepEqual(t, getGrant, newGrant)
-	})
+		tc.expected(t, resp)
+	}
 
-	t.Run("missing required fields", func(t *testing.T) {
-		resp := httptest.NewRecorder()
-		req, err := http.NewRequest(http.MethodPost, "/api/grants", strings.NewReader(`{}`))
-		assert.NilError(t, err)
-		req.Header.Add("Authorization", "Bearer "+adminAccessKey(srv))
-		req.Header.Add("Infra-Version", "0.13.6")
+	testCases := map[string]testCase{
+		"missing required fields": {
+			setup: func(t *testing.T, req *http.Request) {
+				req.Header.Set("Authorization", "Bearer "+adminAccessKey(srv))
+			},
+			body: api.CreateGrantRequest{},
+			expected: func(t *testing.T, resp *httptest.ResponseRecorder) {
+				assert.Equal(t, resp.Code, http.StatusBadRequest, resp.Body.String())
 
-		routes.ServeHTTP(resp, req)
-		assert.Equal(t, resp.Code, http.StatusBadRequest, resp.Body.String())
+				respBody := &api.Error{}
+				err := json.Unmarshal(resp.Body.Bytes(), respBody)
+				assert.NilError(t, err)
 
-		respBody := &api.Error{}
-		err = json.Unmarshal(resp.Body.Bytes(), respBody)
-		assert.NilError(t, err)
+				expected := []api.FieldError{
+					{Errors: []string{"one of (user, group) is required"}},
+					{FieldName: "privilege", Errors: []string{"is required"}},
+					{FieldName: "resource", Errors: []string{"is required"}},
+				}
+				assert.DeepEqual(t, respBody.FieldErrors, expected)
+			},
+		},
+		"success": {
+			setup: func(t *testing.T, req *http.Request) {
+				req.Header.Set("Authorization", "Bearer "+adminAccessKey(srv))
+			},
+			body: api.CreateGrantRequest{
+				User:      someUser.ID,
+				Privilege: models.InfraAdminRole,
+				Resource:  "some-cluster",
+			},
+			expected: func(t *testing.T, resp *httptest.ResponseRecorder) {
+				assert.Equal(t, resp.Code, http.StatusCreated)
 
-		expected := []api.FieldError{
-			{Errors: []string{"one of (user, group) is required"}},
-			{FieldName: "privilege", Errors: []string{"is required"}},
-			{FieldName: "resource", Errors: []string{"is required"}},
-		}
-		assert.DeepEqual(t, respBody.FieldErrors, expected)
-	})
+				expected := jsonUnmarshal(t, fmt.Sprintf(`
+				{
+					"id": "<any-valid-uid>",
+					"created_by": "%[1]v",
+					"privilege": "%[2]v",
+					"resource": "some-cluster",
+					"user": "%[3]v",
+					"created": "%[4]v",
+					"updated": "%[4]v",
+					"wasCreated": true
+				}`,
+					accessKey.IssuedFor,
+					models.InfraAdminRole,
+					someUser.ID.String(),
+					time.Now().UTC().Format(time.RFC3339),
+				))
+				actual := jsonUnmarshal(t, resp.Body.String())
+				assert.DeepEqual(t, actual, expected, cmpAPIGrantJSON)
+			},
+		},
+		"unprivileged support admin grant": {
```

Comment:
Maybe something like:

```suggestion
		"admin can not grant infra support admin role": {
```

Reply:
I've been thinking of the InfraAdminRole as being essentially an OrgAdminRole, and the InfraSupportAdminRole as only able to do high level org manipulation. I think we still want orgs to have to invite a support admin into their role (and grant them admin access) in order to change anything in an org (i.e. principle of least privilege)